tests/test_oauth.py

```diff
@@ -360,26 +365,60 @@ def test_start_flow(self):
             parsed = urlparse.urlparse(url)
             params = urlparse.parse_qs(parsed.query)
 
-            # check parameters
-            assert_equal(
-                params,
-                {
-                    'state': [creds['state']],
-                    'response_type': ['code'],
-                    'client_id': [self.provider.client_id],
-                    'redirect_uri': [
-                        web_url_for('oauth_callback',
-                                    service_name=self.provider.short_name,
-                                    _absolute=True)
-                    ]
-                }
-            )
+            # Check parameters
+            expected_params = {
+                'state': [creds['state']],
+                'response_type': ['code'],
+                'client_id': [self.provider.client_id],
+                'redirect_uri': [
+                    web_url_for(
+                        'oauth_callback',
+                        service_name=self.provider.short_name,
+                        _absolute=True
+                    )
+                ]
+            }
+            assert_equal(params, expected_params)
 
-            # check base URL
-            assert_equal(
-                url.split('?')[0],
-                'https://mock2.com/auth',
-            )
+            # Check base URL
+            assert_equal(url.split('?')[0], 'https://mock2.com/auth')
+
+    def test_start_flow_oauth_no_redirect(self):
+        # Generate the appropriate URL and state token - addons that do not allow `redirect_uri`
+
+        # Temporarily add the mock provider to the `ADDONS_OAUTH_NO_REDIRECT` list
+        ADDONS_OAUTH_NO_REDIRECT.append(self.provider.short_name)
```

Comment:
For this one, I can't find a better way to 1) auto setup/teardown `ADDONS_OAUTH_NO_REDIRECT` and to 2) reduce dup code between `test_start_flow_oauth_standard()` and `test_start_flow_oauth_no_redirect()`.

Reply:
this is fine with me, you're duplicating less code